internal/fieldmask/fieldmask.go

```diff
@@ -0,0 +1,165 @@
+package fieldmask
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/tidwall/gjson"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+const key = "fields"
+
+// FieldMask is recursive structure to define a path mask
+type FieldMask map[string]FieldMask
+
+// New create a FieldMask from the input array of paths. The array should contain JSON paths with dit "." notation.
+func New(paths []string) FieldMask {
+	mask := make(FieldMask)
+	for _, path := range paths {
+		current := mask
```

Comment:
not a blocker comment per se, but curious what you think?

Reply:
I have made this recursive, but changed the name of the method to `Build`, since `New` almost always represent a non-receiver method in most programming language. And in this case we do need a receiver to keep track of the current path. 
I am not sure if this could have been converted to recursive without an extra variable.

Reply:
regardless of that last poing @sayan-biswas (fwiw I don't see another way other than what you did) I don't know what you think, but this version is way more elegant and understandable.

works for me - thanks

Reply:
Agree.